### PR TITLE
feature: dynamically update maintainer list

### LINKED ISSUE #17: [feature request] pouchrobot dynamically recognize maintainer file of github repo instead of hard code in pouchrobot's file

### PR DESCRIPTION
**Pull request by zhuangqh**
Signed-off-by: zhuangqh <zhuangqhc@163.com>

fixes #17

### CODE REVIEW

File: server/utils/maintainer/maintainer.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	var newestMaintainers []string
+	doc.Find("tr").Each(func(i int, s *goquery.Selection) {
+		// First column in table's body.
+		githubID := s.Find("td").First().Text()
+		if githubID != "" {
+			newestMaintainers = append(newestMaintainers, githubID)
+		}
+	})
+
+	maintainers = newestMaintainers
```
Comment (allencloud) [inline server/utils/maintainer/maintainer.go]: If we add a lock in this file, do we have to lock the maintainers variable here? @zhuangqh

Comment (zhuangqh) [reply server/utils/maintainer/maintainer.go]: The maintainers variable is under reader-writer race. Did you mean to move the lock to 
```
mutex.Lock()
maintainers = newestMaintainers
mutex.Unlock()
```
@allencloud

Comment (allencloud) [reply server/utils/maintainer/maintainer.go]: It seems that I misunderstood what you did.

**Comment by allencloud** [comment]
Could you rebase your code to the latest master? @zhuangqh 
I think we need your change. After your code change, I wish to reclaim merge rights from all maintainers, and all merging action will be done by PouchRobot.

**Pull request closed without merging**